### PR TITLE
Lwm2m message handling update

### PR DESCRIPTION
**Pull request by jheiskan81**
Fixed Coap pending entry leak for Conformance message send.
Fixed Notification send blocker.
RD client have own LwM2M message for secure Update and registration message send.

### CODE REVIEW

**Comment by jheiskan81** [comment]
@rlubos @SeppoTakalo @simensrostad  Please review.

File: subsys/net/lib/lwm2m/lwm2m_rd_client.c [hunk 1]
```diff
@@ -127,6 +127,25 @@ struct lwm2m_rd_client_info {
...
+	if (client.rd_message.ctx) {
+		return NULL;
+	}
+
+	client.rd_message.ctx = client.ctx;
+	return &client.rd_message;
+
+}
+
+struct lwm2m_message *lwm2m_get_rd_message(void)
```
Comment (SeppoTakalo) [inline subsys/net/lib/lwm2m/lwm2m_rd_client.c:141]: `rd_get` and `get_rd` are way too close together. It is not clear which does what.
Please refactor either of these.

For example this function could be
```suggestion
struct lwm2m_message *lwm2m_get_ongoing_rd_msg(void)
```

**Review by SeppoTakalo — COMMENTED** [review]
Looks OK. I would just refactor one function name to make a clear distinction.

File: subsys/net/lib/lwm2m/lwm2m_rd_client.c (hunk 1 above)
Comment (jheiskan81) [reply subsys/net/lib/lwm2m/lwm2m_rd_client.c:141]: Ok Thanks for good proposal

**Review by SeppoTakalo — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**